thorlcr/msort/tsorts.cpp

```diff
@@ -100,12 +100,13 @@ class CWriteIntercept : public CSimpleInterface
                 // right create idx
                 StringBuffer tempname;
                 GetTempFilePath(tempname.clear(),"srtidx");
-                idxFile.setown(createIFile(tempname.str()));
-                idxFileIO.setown(idxFile->open(IFOcreaterw));
+                Owned<IFile> tmpFile = createIFile(tempname.str());
+                idxFile.setown(new CFileOwner(tmpFile, activity.queryTempFileSizeTracker()));
```

Comment:
I would be tempted to add a helper function so this could be

```
idxFile.setown(activity.createOwnedTempFile(tmpFile))
```
or even
```
idxFile.setown(activity.createOwnedTempFile(tempname))
```
To clarify the code, reduce duplication and improve encapsulation.

Not for this PR though.